src/database/Database.h

```diff
@@ -285,4 +248,89 @@ class DBTimeExcluder : NonCopyable
     DBTimeExcluder(Application& mApp);
     ~DBTimeExcluder();
 };
+
+// Select a set of records using a client-defined query string, then map
+// each record into an element of a client-defined datatype by applying a
+// client-defined function (the records are accumulated in the "out"
+// vector).
+template <typename T>
+void
+selectMap(Database& db, std::string const& selectStr,
```

Comment:
many of those changes should be squashed into the commit that introduced the change in the first place - this causes churn, making it harder to check what is going on

Reply:
Done.